### DISCUSSION TITLE
Unnecesary calculus (softmax)

### DISCUSSION THREAD

**Discussion by nazarPuriy** [mrdbourke/pytorch-deep-learning#302] [Ideas]
I am not an expert in the field but I think that in several notebooks I noticed unnecessary calculus.
```python
torch.argmax(y_pred, dim=1)
```
the softmax is not needed.

first of all the softmax divides every output by the same number so you can ignore that term when you take the max.

For example, if you have a1,...,an and max ai when you do ca1,...can the maximum will be cai., where c is a real positive number.

Then as the exponential is an injective growing function the maximum of the e^x is at the biggest x. 

So just doing:

```python
torch.argmax(y_pred, dim=1)
```

is enough.

**Comment by mrdbourke** [comment]
yes you're right!

it often is unnecessary if you're just looking for the `torch.argmax()` of prediction logits.

However, for display purposes, I like the scaled outputs of `torch.softmax()`, I find these are easier to interpret as prediction probabilities.

But if you're not displaying them, you're right, you can skip the `torch.softmax()`.
